R/distToCells.R

```diff
@@ -50,86 +52,85 @@
+#'                              metric = "min",
 #'                              img_id = "ImageNb")
 #'
 #' plotSpatial(pancreasSCE,
 #'             img_id = "ImageNb",
 #'             node_color_by = "distToCells",
 #'             scales = "free")
 #'
-#' @author Daniel Schulz (\email{daniel.schulz@@uzh.ch})
+#' @author Daniel Schulz & Bruno Palau (\email{daniel.schulz@@uzh.ch})
 #' @importFrom distances distances distance_columns
-#' @importFrom MatrixGenerics rowMins
+#' @importFrom MatrixGenerics rowMins rowMaxs rowMeans rowMedians
 #' @export
-minDistToCells <- function(object,
-                           x_cells,
-                           img_id,
-                           name = "distToCells",
-                           coords = c("Pos_X","Pos_Y"),
-                           return_neg = TRUE,
-                           BPPARAM = SerialParam()){
-  
-  .valid.minDistToCells.input(object,x_cells,name,coords,img_id,return_neg)
-  
+distToCells <- function (object,
+                        x_cells,
+                        img_id,
+                        name = "distToCells",
+                        coords = c("Pos_X","Pos_Y"),
+                        metric="min",
+                        return_neg = TRUE,
+                        BPPARAM = SerialParam()){
+  .valid.distToCells.input(object, x_cells, name, coords, metric,
+                              img_id, return_neg)
   cur_meta <- metadata(object)
   metadata(object) <- list()
-  
   cur_intmeta <- int_metadata(object)
-  
   object$x_cells <- x_cells
-
   cur_out <- bplapply(
     unique(colData(object)[[img_id]]),
-    function(x){
-      
-      cur_obj <- object[,as.character(colData(object)[[img_id]]) == x]
-      
+    function(x) {
+      cur_obj <- object[, as.character(colData(object)[[img_id]]) == x]
       cur_obj[[name]] <- NA
-      if (sum(cur_obj$x_cells) == 0 | sum(cur_obj$x_cells) == ncol(cur_obj)) {
+      if (sum(cur_obj$x_cells) == 0 | sum(cur_obj$x_cells) ==
+          ncol(cur_obj)) {
         return(cur_obj)
       }
-      
       patch_cells <- which(cur_obj$x_cells)
       non_patch_cells <- which(!cur_obj$x_cells)
-      
       if (is(object, "SpatialExperiment")) {
         dist_mat <- distances(spatialCoords(cur_obj))
-      } else {
-        dist_mat <- distances(as.matrix(colData(cur_obj)[,coords]))
       }
-      
-      pos_dist <- distance_columns(dist_mat,column_indices = patch_cells)
-      dist_to_patch <- rowMins(pos_dist)
-      neg_dist <- distance_columns(dist_mat,column_indices = non_patch_cells)
-      dist_from_patch <- rowMins(neg_dist)
-      
-      # cells that had a 0 distance to the cells of interest can be substitutes
-      # with the negative distances from the cells of interest
-      if(return_neg == TRUE) {
-        dist_to_patch[dist_to_patch == 0] <- -dist_from_patch[dist_to_patch == 0]
+      else {
+        dist_mat <- distances::distances(as.matrix(colData(cur_obj)[,
```

Comment:
I think you do not need to specify the package here, since you import it above > distances(as.matrix(colData(cur_obj)[,coords])) should be enough